src/index.js

```diff
@@ -114,7 +114,7 @@ async function build(options, stageDir) {
     await options.sign(setupPath)
   }
   if (options.msi && process.platform === 'win32') {
-    const outFile = options.msiExe || `${metadata.productName}Setup.msi`;
+    const outFile = options.msiExe || `${metadata.productName || metadata.name}Setup.msi`;
```

Comment:
If neither msiExe or productName are specified, then the MSI is called `undefinedSetup.msi`. This fixes the issue as the package name is mandatory.